src/Windowing/Silk.NET.Windowing.Sdl/SdlWindow.cs

```diff
@@ -0,0 +1,426 @@
+                : _extendedOptionsCache.WindowState;
+            set
+            {
+                _extendedOptionsCache.WindowState = value;
+                if (!IsInitialized)
+                {
+                    return;
+                }
+
+                switch (value)
+                {
+                    case WindowState.Normal:
+                    {
+                        Sdl.RestoreWindow(SdlWindow);
+                        break;
+                    }
+                    case WindowState.Minimized:
+                    {
+                        Sdl.MinimizeWindow(SdlWindow);
+                        break;
+                    }
+                    case WindowState.Maximized:
+                    {
+                        Sdl.MaximizeWindow(SdlWindow);
+                        break;
+                    }
+                    case WindowState.Fullscreen:
+                    {
+                        Sdl.SetWindowFullscreen(SdlWindow, (uint) WindowFlags.WindowFullscreen);
+                        break;
+                    }
+                    default:
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(value), value, null);
+                    }
+                }
+            }
+        }
+
+        public WindowBorder WindowBorder
+        {
+            get => IsInitialized
+                ? _extendedOptionsCache.WindowBorder = ToWindowBorder((WindowFlags) Sdl.GetWindowFlags(SdlWindow))
+                : _extendedOptionsCache.WindowBorder;
+            set
+            {
+                _extendedOptionsCache.WindowBorder = value;
+                if (!IsInitialized)
+                {
+                    return;
+                }
+
+                switch (value)
+                {
+                    case WindowBorder.Resizable:
+                    {
+                        Sdl.SetWindowBordered(SdlWindow, SdlBool.True);
+                        Sdl.SetWindowResizable(SdlWindow, SdlBool.True);
+                        break;
+                    }
+                    case WindowBorder.Fixed:
+                    {
+                        Sdl.SetWindowBordered(SdlWindow, SdlBool.True);
+                        Sdl.SetWindowResizable(SdlWindow, SdlBool.False);
+                        break;
+                    }
+                    case WindowBorder.Hidden:
+                    {
+                        Sdl.SetWindowBordered(SdlWindow, SdlBool.False);
+                        Sdl.SetWindowResizable(SdlWindow, SdlBool.False);
+                        break;
+                    }
+                    default:
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(value), value, null);
+                    }
+                }
+            }
+        }
+
+        public bool TransparentFramebuffer => false; // doesn't look like SDL doesn't support this
```

Comment:
not sure what you mean here? Seems like SDL does support transparent framebuffers, but I guess that's what the comment suggests too 😛

Reply:
Unfortunately it doesn't, at least not in the same way GLFW does. Full window transparency is only documented to have been achieved with SDL by manually tapping into user32 and using SetLayeredWindowAttributes